src/cmd/alias.go

```diff
@@ -40,9 +40,13 @@ Note that this can conflict with other tools that also define additional Git com
 		}
 	},
 	PreRunE: func(cmd *cobra.Command, args []string) error {
+		arg := ""
+		if len(args) > 0 {
+			arg = args[0]
+		}
```

Comment:
I'm not a fan of this as this is replicating a bit of what `validateArgsCountFunc` is doing. We do need to prevent accessing the first element until after we have validated the count but believe we could that with the following:
```go
return util.FirstError(
   validateArgsCountFunc(args, 1),
   func() error {
     return validateBooleanArgument(args[0])
   },
}
```

Reply:
Honestly, I find the entire indirection of errors to be less clear than it needs to be. The function would be more succinctly conveyed along the lines of:

```
PreRunE: func(cmd *cobra.Command, args []string) error {
	if len(args) < 1 {
		return errors.New("Too few arguments")
	}
	if len(args) > 1 {
		return errors.New("Too many arguments")
	}
	if isBooleanArgument(args[0]) {
		return fmt.Errorf("Invalid value: '%s'", args[0])
	}
	return nil
},
```

Anyway, I am sure this is in good hands.

Reply:
Yeah we may have taken the indirection a bit far. That is essentially what we had before and then tried to dry up the duplication of validating the number of arguments with:
```go
err := validateArgsCount(args, 1)
if err != nil {
  return err 
}
return validateBooleanArgument(args[0])
```
Then we moved to util.FirstError but in an ongoing effort to remove  `if err != nil` checks